container/rules/docker_toolchains.bzl

```diff
@@ -91,45 +92,19 @@ def _language_tool_layer_impl(
         # debs to installables_tars.
 
     elif packages != []:
-        # Prepare base image for the download_pkgs rule.
-        download_base = ctx.files.base[0]
-
-        # Create an intermediate image with additional gpg keys used to download packages.
-        if keys != []:
-            image_with_keys = "%s_with_keys" % ctx.attr.name
-
-            # Declare intermediate output file generated by add_apt_key rule.
-            image_with_keys_output_executable = ctx.actions.declare_file(image_with_keys)
-            image_with_keys_output_tarball = ctx.actions.declare_file(image_with_keys + ".tar")
-            image_with_keys_output_layer = ctx.actions.declare_file(image_with_keys + "-layer.tar")
-
-            _key.implementation(
-                ctx,
-                name = image_with_keys,
-                image_tar = ctx.files.base[0],
-                keys = keys,
-                output_executable = image_with_keys_output_executable,
-                output_tarball = image_with_keys_output_tarball,
-                output_layer = image_with_keys_output_layer,
-            )
-            download_base = image_with_keys_output_tarball
-
-        # Declare intermediate output file generated by download_pkgs rule.
-        download_pkgs_output_executable = ctx.actions.declare_file(ctx.attr.name + "-download_pkgs_output_executable.sh")
-        download_pkgs_output_tar = ctx.actions.declare_file(ctx.attr.name + "-download_pkgs_output_tar.tar")
-        download_pkgs_output_script = ctx.actions.declare_file(ctx.attr.name + "-download_pkgs_output_script.sh")
-
-        # download_pkgs rule consumes 'packages' and 'additional_repos'.
-        _download.implementation(
+        download_pkgs_output_tar = ctx.attr.name + "-download_pkgs_output_tar.tar"
+        download_pkgs_output_script = ctx.attr.name + "-download_pkgs_output_script.sh"
+
+        aggregated_debian_tar = _aggregate_debian_pkgs.implementation(
```

Comment:
you should be calling _generate_deb_tar here, not _aggregate_debian_pkgs as here you no longer have layers, you have packages

Reply:
Yea, the naming here is confusing and will change it. Thanks for pointing out.